sdk/cosmos/azure-cosmos/src/main/java/com/azure/cosmos/GatewayConnectionConfig.java

```diff
@@ -119,18 +120,26 @@ public InetSocketAddress getProxy() {
      * @return the {@link GatewayConnectionConfig}.
      */
 
-    public GatewayConnectionConfig setProxy(InetSocketAddress proxy) {
-        this.inetSocketProxyAddress = proxy;
+    public GatewayConnectionConfig setProxy(ProxyOptions proxy) {
+        if (proxy.getType() != ProxyOptions.Type.HTTP) {
+            throw new IllegalArgumentException("Only http proxy type is supported.");
```

Comment:
as we don't support password with proxy, shouldn't we throw if password is set?

Reply:
I am not sure. may be we should, may be we should not, it will just get ignored. 

But we definitely should add more documentation to setProxy() API that we only support HTTP proxy as of now - without any username and password. 

I see, `setProxy()` docs are incorrect and should be updated. 

@xinlian12 - please create another PR for this - with updated docs - "that we only support  HTTP proxy type with just the routing address. Username and password are not supported"

Reply:
Thanks~ have created another PR for the docs update:
chttps://github.com/Azure/azure-sdk-for-java/pull/11672